src/platforms/web/runtime/modules/dom-props.js

```diff
@@ -44,7 +44,7 @@ function updateDOMProps (oldVnode: VNodeWithData, vnode: VNodeWithData) {
       // non-string values will be stringified
       elm._value = cur
       // avoid resetting cursor position when value is the same
-      const strCur = isUndef(cur) ? '' : String(cur)
+      const strCur = isNumNegZero(cur) ? '-0' : isUndef(cur) ? '' : String(cur)
```

Comment:
I think in the interest of least impactful, this collection of ternaries should keep the `isUndef()` at the top like so:

```js
const strCur = isUndef(cur) ? '' : isNumNegZero(cur) ? '-0' : String(cur)
```

This keeps the case of `cur` being empty as the fastest/shortest path.